nntrainer/layers/activation_layer.h

```diff
@@ -83,61 +87,11 @@ class ActivationLayer : public Layer {
   inline static const std::string type = "activation";
 
 private:
-  ActiFunc
-    acti_func; /**< activation function designating the activation operation */
+  using PropTypes = std::tuple<props::Activation>;
 
-  /**
-   * @brief setActivation by custom activation function
-   * @note  apply derivative as this activation_prime_fn does not utilize
-   * derivative
-   * @param[in] std::function<Tensor(Tensor const &, Tensor &)> activation_fn
-   * activation function to be used
-   * @param[in] std::function<Tensor(Tensor const &, Tensor &)>
-   * activation_prime_fn activation_prime_function to be used
-   * @retval #ML_ERROR_NONE when successful
-   */
-  int setActivation(
-    std::function<Tensor &(Tensor const &, Tensor &)> const &activation_fn,
-    std::function<Tensor &(Tensor &, Tensor &)> const &activation_prime_fn);
+  std::unique_ptr<PropTypes> activation_props; /**< activation props */
```

Comment:
is this needed now? as layer_node caches the activation property and getActivationType is returned by layer_node itself.

Reply:
This is needed to support below format.

[relu1]
type=activation
activation=relu

Reply:
the `activation` can be added by layer node just like for every other layer.

Reply:
In that case, we will need dynamic_cast to activation layer because we have to propagate activation somehow to the actual layer.